src/no_std.rs

```diff
@@ -210,22 +249,29 @@ pub(crate) struct ListGuard<'a, T> {
 
     /// The locked list.
     pub(crate) guard: Option<MutexGuard<'a, ListenerSlab<T>>>,
+
+    /// Tasks to wake up once this guard is dropped.
+    tasks: Vec<Task>,
 }
 
 impl<T> ListGuard<'_, T> {
     #[cold]
-    fn process_nodes_slow(
-        &mut self,
-        start_node: Node<T>,
-        tasks: &mut Vec<Task>,
-        guard: &mut MutexGuard<'_, ListenerSlab<T>>,
-    ) {
+    fn process_nodes_slow(&mut self, start_node: Node<T>) {
         // Process the start node.
-        tasks.extend(start_node.apply(guard));
+        self.tasks
+            .extend(start_node.apply(self.guard.as_mut().unwrap()));
```

Comment:
why can't we hold onto `self.guard.as_mut().unwrap()`? (theoretically, the unnecessary branching should get optimized away, but still, it feels unclean.)